src/figuresClasses.ts

```diff
@@ -1,19 +1,119 @@
 export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle',
+  color: 'red' | 'green' | 'blue',
+}
+
+interface ITriangle extends Figure {
+  side1: number,
+  side2: number,
+  side3: number,
+}
 
+interface ICircle extends Figure {
+  radius: number
 }
 
-export class Triangle {
+interface IRectangle extends Figure {
+  width: number,
+  height: number,
+}
 
+function throwErrorOnNegativeValue(variable: number): Error | void {
+  if (variable <= 0) {
+    throw new Error('This side can\'t be less than 0');
+  }
 }
 
-export class Circle {
+const throwErrorTriangle = (): Error => {
+  throw new Error(
+    'This triangle does not exist. Sum of 2 sides must be more than 3rd side',
+  );
+};
+
+export class Triangle implements ITriangle {
+  readonly shape: 'triangle' | 'circle' | 'rectangle';
+
+  constructor(
+    readonly color: 'red' | 'green' | 'blue',
+    readonly side1: number,
+    readonly side2: number,
+    readonly side3: number,
+  ) {
+    throwErrorOnNegativeValue(side1);
+    throwErrorOnNegativeValue(side2);
+    throwErrorOnNegativeValue(side3);
+
+    if (!(side1 + side2 > side3)) {
+      throwErrorTriangle();
+    }
+
+    if (!(side2 + side3 > side1)) {
+      throwErrorTriangle();
+    }
 
+    if (!(side1 + side3 > side2)) {
+      throwErrorTriangle();
+    }
```

Comment:
You do the same action for each case. Just combine the conditions by `||` operator